### ISSUE TITLE
Calories not syncing from Strava to Garmin Connect

### ISSUE THREAD

**Issue by prueker** [cpfair/tapiriik#353]
Everything else including power, heartrate is syncing, but the calories are 0.

**Comment by sergmelikyan** [comment]
I confirm this issue, Calories are not synced to the Garmin Connect, but I think this is caused by the broken export (from Strava?) rather than import - tcx files which I use for exporting to Dropbox also doesn't contain this information:
`<Calories>0</Calories>`

**Comment by sergmelikyan** [comment]
@prueker are you also exporting from Strava to Garmin Connect? Or some other source?

@cpfair anything else we can do to help with this issue?

**Comment by sergmelikyan** [comment]
According to strava this value is provided in a field calories: http://strava.github.io/api/v3/activities/ which we actually read in https://github.com/AndrejWalk/tapiriik/blob/master/tapiriik/services/Strava/strava.py#L176

**Comment by prueker** [comment]
@sergmelikyan Yes my issue is sync from Strava to Garmin Connect, as I said all other fields do sync, but the calories are 0.

**Comment by sergmelikyan** [comment]
Looks like problem is on the side of reading activities from the Strava, in [strava.py on line 130](https://github.com/cpfair/tapiriik/blob/master/tapiriik/services/Strava/strava.py#L130) we are doing request to Strava to read all activities for the given athlete like this:

`https://www.strava.com/api/v3/athletes/<user-id>/activities`

Which according to Strava API Reference [here](https://strava.github.io/api/v3/activities/#get-activities) doesn't contain "calories" field for individual activity, only "kilojoules" for rides and nothing for runs for example. 

My experiments confirm that output data from this requests doesn't contain this field. I have create new API app on [this](https://www.strava.com/settings/api) page and used provided Access Token to execute following cUrl:

`curl -G https://www.strava.com/api/v3/athletes/9535682/activities -d access_token=<my-access-token> > output.json`

Now question is how to resolve this? Here are my thoughts:
1) Use different API call to get information about activity which has "calories" field
2) Calculate calories by ourselves
3) Reach out to Strava developers and ask to include this field or help to workaround this

**Method 1: Different API Call**
I spent around 20m reading through Strava API Reference and seems like the only API call which return this value is  

`https://www.strava.com/api/v3/activities/<activity-id>`

Which means that we will have to do instead 1 API call to Strava x +1 calls where x is number of activities to sync. This will significantly increase time and load on the system as well as overall API call numbers which are limited.

@cpfair what do you think about load which will be generated if we will switch to request per activity model?

**Method 2: Calculate value by ourselves**
Right now it seems impossible to calculate this number in the same way as Strava does, I was not able to find algorithm which we can replicate based on the data provided by following call:

`https://www.strava.com/api/v3/athletes/<user-id>/activities`

**Method 3: Reach out to Strava developers**
I have reached out to Strava developers via e-mail asking how we can deal with this, as well as want to mention @stravadk in hopes he can provide some feedback here.

References:
- Strava API Reference: [Activities](https://strava.github.io/api/v3/activities/) 
- Output 1: List of activities for given athlete (me) - attached [output-1.txt](https://github.com/cpfair/tapiriik/files/1178779/output-1.txt)
- Output 2: Specific activity (my last ride) - attached [output-2.txt](https://github.com/cpfair/tapiriik/files/1178780/output-2.txt)

P.S. Seems like providing <user-id> is In the current API call to list activities is excessive and can be safely dropped:

`https://www.strava.com/api/v3/athletes/<user-id>/activities`

**Comment by sergmelikyan** [comment]
**Method 3: Reach out to Strava developers**

I have received response stating that somewhere in the beginning of the next year "calories" value might get more exposure, but right now the only way to get this value is using aforementioned call to get activity details.

**Comment by CASNetlink** [comment]
It looks like this is unchanged in April 2022 - calories are not included in API v3 calls to https://www.strava.com/api/v3/athlete/activities
